hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/HFileOutputFormat2.java

```diff
@@ -653,10 +664,9 @@ static void configureIncrementalLoad(Job job, List<TableInfo> multiTableInfo,
 
     for (TableInfo tableInfo : multiTableInfo) {
       regionLocators.add(tableInfo.getRegionLocator());
-      String tn = writeMultipleTables
-        ? tableInfo.getRegionLocator().getName().getNameWithNamespaceInclAsString()
-        : tableInfo.getRegionLocator().getName().getNameAsString();
-      allTableNames.add(tn);
+      allTableNames.add(writeToTableWithNamespace ?
```

Comment:
do we want this to be `writeMultipleTables && writeToTableWithNamespace` ?

Reply:
Just thinking that when we upgrade to 3.0, the code will be:

```
 String tn = writeMultipleTables
        ? tableInfo.getRegionLocator().getName().getNameWithNamespaceInclAsString()
        : tableInfo.getRegionLocator().getName().getNameAsString();
```

and our new config will disappear. So if someone enabled writeToTableWithNamespace in hbase2 (but not writeMultipleTables) and they go to 3.0, they'll stop writing with the namespace included.

Reply:
Sounds right to me.

Reply:
Actually in 3.0, FWIW it is always withNamespace. Do you think it makes any difference at all with or without the change you suggested?

Reply:
Yea I think we need the code I suggested. In 2.0, we only honor the new config if writeMultipleTables is true. That's what we documented in the comment on the config key. That's also how the behavior works in the other block in this file:

```
if (writeMultipleTables) {
  // some stuff
  tableNameBytes = writeToTableWithNamespace ? one : two;
}
```

We need to do the same thing here, so:

```

 String tn = writeMultipleTables && writeToTableWithNamespace
        ? tableInfo.getRegionLocator().getName().getNameWithNamespaceInclAsString()
        : tableInfo.getRegionLocator().getName().getNameAsString();

```

That way we are consistent. Otherwise let's say someone sets writeToTableWithNamespace to true in 2.0, but not writeMultipleTables. When they upgrade to 3.0, the logic will look like this:

```

String tn = writeMultipleTables
        ? tableInfo.getRegionLocator().getName().getNameWithNamespaceInclAsString()
        : tableInfo.getRegionLocator().getName().getNameAsString();
```

So since they didn't set writeMultipleTables to true, the output path would revert to not include namespace. 

Does that make sense? I think we just need the && so we are consistent in this file and with 3.0.

Reply:
Got it. changing.